WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/watchface/ColorStyleIdAndResourceIds.kt

```diff
@@ -0,0 +1,167 @@
+    )
+
+    object Green : ColorStyleIdAndResourceIds(
+        id = GREEN_COLOR_STYLE_ID,
+        name = GREEN_COLOR_STYLE_NAME_RESOURCE_ID,
+        iconResourceId = GREEN_COLOR_STYLE_ICON_ID,
+        complicationStyleDrawableId = R.drawable.complication_green_style,
+        primaryColorId = R.color.green_primary_color,
+        secondaryColorId = R.color.green_secondary_color,
+        backgroundColorId = R.color.green_background_color,
+        outerElementColorId = R.color.green_outer_element_color
+    )
+
+    object Blue : ColorStyleIdAndResourceIds(
+        id = BLUE_COLOR_STYLE_ID,
+        name = BLUE_COLOR_STYLE_NAME_RESOURCE_ID,
+        iconResourceId = BLUE_COLOR_STYLE_ICON_ID,
+        complicationStyleDrawableId = R.drawable.complication_blue_style,
+        primaryColorId = R.color.blue_primary_color,
+        secondaryColorId = R.color.blue_secondary_color,
+        backgroundColorId = R.color.blue_background_color,
+        outerElementColorId = R.color.blue_outer_element_color
+    )
+
+    object White : ColorStyleIdAndResourceIds(
+        id = WHITE_COLOR_STYLE_ID,
+        name = WHITE_COLOR_STYLE_NAME_RESOURCE_ID,
+        iconResourceId = WHITE_COLOR_STYLE_ICON_ID,
+        complicationStyleDrawableId = R.drawable.complication_white_style,
+        primaryColorId = R.color.white_primary_color,
+        secondaryColorId = R.color.white_secondary_color,
+        backgroundColorId = R.color.white_background_color,
+        outerElementColorId = R.color.white_outer_element_color
+    )
+
+    companion object {
+        /**
+         * Translates the string id to the correct ColorStyleIdAndResourceIds object.
+         */
+        fun getColorStyleConfig(id: String): ColorStyleIdAndResourceIds {
+            return when (id) {
+                Ambient.id -> Ambient
+                Red.id -> Red
+                Green.id -> Green
+                Blue.id -> Blue
+                White.id -> White
+                else -> White
+            }
+        }
+
+        /**
+         * Returns a list of [UserStyleSetting.ListUserStyleSetting.ListOption] for all sealed
+         * ColorStyleIdAndResourceIds objects. The watch face settings APIs use this to set up
+         * options for the user to select a style.
+         *
+         * IMPORTANT: To avoid reflection, the full list of ColorStyleIdAndResourceIds are manually
+         * included below (you need to use reflection to get full list of sealed classes which I did
+         * not want to do). If you add a new style, add it to the list below. Otherwise, it will not
+         * show up to the user!
+         */
+        fun toOptionList(context: Context): List<ListUserStyleSetting.ListOption> {
+            // Hardcoded list of all ColorStyleIdAndResourceIds. If you add a new one, add it here
+            // as well.
+            val colorStyleIdAndResourceIdsList = listOf(Red, Green, Blue, White, Ambient)
```

Comment:
Could we make `ColorStyleIdAndResourceIds` an `enum class`? Then we could use `values()` or `enumValues()`, to avoid having to remember to update this list.

I am also quite partial to a certain library that allows generating these lists of `sealed class`s with only `object`s 😸

Reply:
Yeah, that makes sense.  Good bye sealed class... :'(